hugolib/site_new.go

```diff
@@ -558,3 +559,9 @@ func (s *Site) AllPages() page.Pages {
 func (s *Site) AllRegularPages() page.Pages {
 	return s.h.RegularPages()
 }
+
+// Translate returns a translated string for id.
+func (s *Site) Translate(id string, args ...any) (string, error) {
+	ctx := context.Background()
+	return lang.New(s.Deps, langs.GetTranslator(s.Language())).Translate(ctx, id, args...)
```

Comment:
The above is propbably simpler to reason about when moving the method, but we do not want to create a new language for each translated string.